client/src/components/categories/CategoryList.jsx

```diff
@@ -1,15 +1,36 @@
-import { useEffect, useState } from "react";
+import { useDebugValue, useEffect, useState } from "react";
 import { getAllCategories } from "../../managers/categoryManager";
 import { Button, ButtonToolbar, Card, CardBody, CardTitle, Table } from "reactstrap";
 import { Link } from "react-router-dom";
+import { deleteCategory } from "../../managers/CategoryManager";
+import ConfirmDeleteModal from "../modals/ConfirmDeleteModal";
 
 export default function CategoryList() {
     const [categories, setCategories] = useState([]);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [categoryIdToDelete, setCategoryIdToDelete] = useState(null);
 
     useEffect(() => {
         getAllCategories().then(setCategories);
     }, []);
 
+    const toggleModal = () => {
+        setIsModalOpen(!isModalOpen);
+    }
+
+    const handleDeleteBtnClick = (id) => {
+        setCategoryIdToDelete(id);
+        toggleModal();
+    };
+
+    const handleDelete = () => {
```

Comment:
Would it make more sense for this to be called `confirmDelete`? Lets discuss as a team the standard nomenclature for this